src/drivers/esp8266_drv.c

```diff
@@ -462,8 +451,14 @@ static void init_wifi()
          * before init has completed b/c this invokes no modem calls.
          * This should only fail if there was an issue with space or config.
          */
-        if (!esp8266_register_ipd_cb(rx_data_cb))
-                pr_error(LOG_PFX "Failed to register IPD callback\r\n");
+        const struct esp8266_event_hooks hooks = {
+                .client_state_changed_cb = client_state_changed_cb,
+                .socket_state_changed_cb = socket_state_changed_cb,
+                .data_received_cb = rx_data_cb,
+        };
+
+        if (!esp8266_register_callbacks(&hooks))
+                pr_warning(LOG_PFX "Failed to register callbacks\r\n");
```

Comment:
If this fails then there may be some undefined behavior, but in all I would not set it at error level.  The system is not necessarily unstable.  But its a warning we should see.